lib/oban.ex

```diff
@@ -1100,4 +1103,258 @@ defmodule Oban do
       raise ArgumentError, "queue #{inspect(queue)} does not exist locally"
     end
   end
+
+  @doc """
+  It creates a facade for the `Oban` functions.
+
+  It allows you to avoid having to pass the `t:Oban.name/0` value to the `Oban` functions, as it is automatically set to
+  the module name. As well as allowing you to configure the `Oban` instance in the application's configuration using
+  the module name under a given OTP application key.
+
+  ## Examples
+
+  ### In an application module
+
+      defmodule MyApp.Oban do
+        use Oban.Instance,
+          otp_app: :my_app,
+          repo: MyApp.Repo
+      end
+
+  Now you can register the `MyApp.Oban` module in the application's supervision tree:
+
+      defmodule MyApp.Application do
+        use Application
+
+        def start(_type, _args) do
+          children = [
+            MyApp.Repo,
+            MyApp.Oban
+          ]
+
+          opts = [strategy: :one_for_one, name: MyApp.Supervisor]
+          Supervisor.start_link(children, opts)
+        end
+      end
+
+  ### Avoiding the need of passing the `Oban` instance
+
+  Instead of calling `Oban` functions passing the `Oban` instance, you can use the `Oban.Instance` module directly:
+
+      Oban.Instance.insert(MyApp.MyJob.new(args: %{ "field" => "value" }))
+  """
+  defmacro __using__(opts \\ []) do
+    {opts, child_opts} = Keyword.split(opts, [:otp_app])
+    otp_app = Keyword.fetch!(opts, :otp_app)
+
+    # credo:disable-for-next-line Credo.Check.Refactor.LongQuoteBlocks
+    quote do
+      @doc """
+      Returns the Oban child spec for the application. This should be added to the application's supervision tree.
+
+      The `:name` option is ignored and set to `#{inspect(__MODULE__)}`.
+
+      ## Examples
+
+          defmodule MyApp.Application do
+            use Application
+
+            children = [
+              {#{inspect(__MODULE__)}, prefix: "special"}
+            ]
+
+            opts = [strategy: :one_for_one, name: MyApp.Supervisor]
+            Supervisor.start_link(children, opts)
+          end
+      """
```

Comment:
I don't feel enough _applications_ generate documentation for us to include `@doc` or `@spec` for generated functions. Some of the specs such as `insert_all` are already complicated and difficult to verify exhaustively, and I'd rather not duplicate them.